types/rlp_marshal.go

```diff
@@ -191,13 +193,19 @@ func (t *Transaction) MarshalRLPTo(dst []byte) []byte {
 func (t *Transaction) MarshalRLPWith(arena *fastrlp.Arena) *fastrlp.Value {
 	vv := arena.NewArray()
 
+	// Specify zero chain ID as per spec.
+	// This is needed to be compatible with other EVM chains and have the same format.
+	if t.Type == DynamicFeeTx {
```

Comment:
What is this new Type of transaction?

Reply:
Moved to https://github.com/0xPolygon/polygon-edge/pull/1237